tx.go

```diff
@@ -189,6 +190,62 @@ func NewTxFromStream(b []byte) (*Tx, int, error) {
 	return &t, offset, nil
 }
 
+// NewTxFromReader creates a transaction from an io.Reader
+func NewTxFromReader(r io.Reader) (*Tx, error) {
```

Comment:
Is there any reason we are building it this way rather than just reading r into a bytes.Buffer and passing that to NewTxFromBytes? Seems like a lot of code duplication